src/intTest/java/uk/nhs/digital/nhsconnect/nhais/IntegrationTestsExtension.java

```diff
@@ -31,21 +34,24 @@ public void beforeEach(ExtensionContext context) {
 
         var jmsTemplate = applicationContext.getBean(JmsTemplate.class);
 
-        var meshInboundQueueName = applicationContext.getEnvironment().getProperty("nhais.amqp.meshInboundQueueName");
-        var meshOutboundQueueName = applicationContext.getEnvironment().getProperty("nhais.amqp.meshOutboundQueueName");
-        var gpSystemInboundQueueName = applicationContext.getEnvironment().getProperty("nhais.amqp.meshInboundQueueName");
+        var meshInboundQueueName = Objects.requireNonNull(
+            applicationContext.getEnvironment().getProperty("nhais.amqp.meshInboundQueueName"));
+        var meshOutboundQueueName = Objects.requireNonNull(
+            applicationContext.getEnvironment().getProperty("nhais.amqp.meshOutboundQueueName"));
+        var gpSystemInboundQueueName = Objects.requireNonNull(
+            applicationContext.getEnvironment().getProperty("nhais.amqp.meshInboundQueueName"));
 
         var receiveTimeout = jmsTemplate.getReceiveTimeout();
         jmsTemplate.setReceiveTimeout(RECEIVE_TIMEOUT_NO_WAIT);
-        while (jmsTemplate.receive(Objects.requireNonNull(meshInboundQueueName)) != null) {
-            LOGGER.info("Purged " + meshInboundQueueName + " message");
-        }
-        while (jmsTemplate.receive(Objects.requireNonNull(meshOutboundQueueName)) != null) {
-            LOGGER.info("Purged " + meshOutboundQueueName + " message");
-        }
-        while (jmsTemplate.receive(Objects.requireNonNull(gpSystemInboundQueueName)) != null) {
-            LOGGER.info("Purged " + gpSystemInboundQueueName + " message");
-        }
+        List.of(meshInboundQueueName, meshOutboundQueueName, gpSystemInboundQueueName)
+            .stream()
+            .map(queueName -> List.of(queueName, DLQ_PREFIX + queueName))
+            .flatMap(Collection::stream)
+            .forEach(queueName -> {
+                while (jmsTemplate.receive(queueName) != null) {
+                    LOGGER.info("Purged '" + queueName + "' message");
+                }
+            });
```

Comment:
This should be enclosed in method like `void purgeQueues(List<String> queueNames)`